nat-lab/run_local.py

```diff
@@ -99,7 +99,12 @@ def main() -> int:
 
 
 def run_build_command(operating_system, args):
-    command = ["../../ci/build.sh", "--default", operating_system]
+    command = [
+        "../../ci/build.sh",
+        "--default",
+        operating_system,
+        "--uniffi-test-bindings",
+    ]
```

Comment:
This fails for me:
```
$ ./run_local.py
|EXECUTE| ../../ci/build.sh --default linux --uniffi-test-bindings
+ RESTART_CONTAINER=--default
+++ dirname ../../ci/build.sh
++ cd ../../ci
++ pwd
+ SCRIPT_DIR=/Users/tomaszklak/Development/libtelio-build/ci
+ BASE_DIR=/Users/tomaszklak/Development/libtelio-build/ci/..
+ [[ 3 -gt 0 ]]
+ case "$1" in
+ set_defaults linux
++ uname -m
+ BUILD_ARCH=arm64
+ BUILD_DEBUG=
+ BUILD_MOOSE=
+ case "$1" in
+ CONTAINER_NAME=libtelio_linux_builder
+ BUILD_CMD='libtelio/ci/build_libtelio.py build linux arm64'
+ [[ arm64 == \a\a\r\c\h\6\4 ]]
+ BUILD_IMAGE=ghcr.io/nordsecurity/build-linux-rust1.72.1:v0.0.1
+ shift
+ shift
+ [[ 1 -gt 0 ]]
+ case "$1" in
+ echo 'Unrecognized parameter '\''--uniffi-test-bindings'\'''
Unrecognized parameter '--uniffi-test-bindings'
+ print_help
+ echo 'Usage: ci/build.sh [--default <android|linux|windows|derpcli|interderpcli>] [--container-name <name>] [--cmd <cmd>] [--image <image>] [--arch <arch>] [--restart] [--debug]'
Usage: ci/build.sh [--default <android|linux|windows|derpcli|interderpcli>] [--container-name <name>] [--cmd <cmd>] [--image <image>] [--arch <arch>] [--restart] [--debug]
+ exit 1
Traceback (most recent call last):
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 196, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 76, in main
    run_build_command("linux", args)
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 108, in run_build_command
    run_command(command)
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 22, in run_command
    subprocess.check_call(command, env=env)
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['../../ci/build.sh', '--default', 'linux', '--uniffi-test-bindings']' returned non-zero exit status 1.
```

Reply:
you might be on the wrong libtelio-build ref

Reply:
It says it should be 1.0.3: https://github.com/NordSecurity/libtelio/blob/LLT-3475_migrate_natlab_tests_to_uniffi_bindings/.github/workflows/gitlab.yml#L34 and I'm on that libtelio-build version.

Reply:
I guess it wasn't updated on libtelio side :smile: use `origin/LLT-3475_migrate_natlab_tests_to_uniffi_bindings` as libtelio-build ref and you should be fine.

Reply:
Switching libtelio-build doesn't really help:

```
################################################################################
Project root (/Users/tomaszklak/Development/libtelio-build) and 'triggered-ref' in ../.github/workflows/gitlab.yml differ
        'cabf069d021d0183e95069ea0617a9ce0ecc72ec' vs 'v1.0.3'
!!! Nat-lab might not behave correctly !!!
################################################################################
|EXECUTE| ../../ci/build.sh --default linux --uniffi-test-bindings
+ RESTART_CONTAINER=--default
+ RESTART_UNIFFI_CONTAINER=--default
+++ dirname ../../ci/build.sh
++ cd ../../ci
++ pwd
+ SCRIPT_DIR=/Users/tomaszklak/Development/libtelio-build/ci
+ BASE_DIR=/Users/tomaszklak/Development/libtelio-build/ci/..
+ [[ 3 -gt 0 ]]
+ case "$1" in
+ set_defaults linux
+ BUILD_OS=linux
++ uname -m
+ BUILD_ARCH=arm64
+ BUILD_DEBUG=
+ BUILD_MOOSE=
+ GEN_UNIFFI_PYTHON_BINDINGS=false
+ case "$1" in
+ CONTAINER_NAME=libtelio_linux_builder
+ BUILD_CMD='libtelio/ci/build_libtelio.py build linux arm64'
+ [[ arm64 == \a\a\r\c\h\6\4 ]]
+ BUILD_IMAGE=ghcr.io/nordsecurity/build-linux-rust1.72.1:v0.0.1
+ shift
+ shift
+ [[ 1 -gt 0 ]]
+ case "$1" in
+ GEN_UNIFFI_PYTHON_BINDINGS=true
+ shift
+ [[ 0 -gt 0 ]]
+ '[' -z libtelio_linux_builder ']'
+ BUILD_WORKDIR=/libtelio-build
+ docker ps
+ grep libtelio_linux_builder
+ RESTART_CONTAINER=--restart
+ [[ -z '' ]]
+ INTERACTIVE=-it
+ [[ darwin22 == \d\a\r\w\i\n* ]]
++ echo /Users/tomaszklak/.ssh
+ SSH_AUTH_SOCK_ARGUMENT='-v /Users/tomaszklak/.ssh:/root/.ssh:ro'
+ GIT_SSH_COMMAND='ssh -o StrictHostKeyChecking=no'
+ [[ true == true ]]
+ case "$BUILD_OS" in
+ BUILD_CMD='libtelio/ci/build_libtelio.py build linux arm64 uniffi-test-bindings'
+ UNIFFI_CONTAINER_NAME=uniffi-generators
+ UNIFFI_CONTAINER_IMAGE=ghcr.io/nordsecurity/uniffi-generators:v0.25.0-3
+ UNIFFI_COMMAND='uniffi-bindgen generate src/libtelio.udl --language python'
+ docker ps
+ grep uniffi-generators
+ RESTART_UNIFFI_CONTAINER=--restart
+ [[ --restart == \-\-\r\e\s\t\a\r\t ]]
+ docker container kill uniffi-generators
Error response from daemon: Cannot kill container: uniffi-generators: No such container: uniffi-generators
+ echo 'Failed to kill container uniffi-generators!'
Failed to kill container uniffi-generators!
+ docker container rm uniffi-generators
Error response from daemon: No such container: uniffi-generators
+ echo 'Failed to remove container uniffi-generators!'
Failed to remove container uniffi-generators!
+ docker run --name=uniffi-generators -d -v /Users/tomaszklak/Development/libtelio-build/ci/..://libtelio-build --workdir=/libtelio-build/libtelio ghcr.io/nordsecurity/uniffi-generators:v0.25.0-3 sleep infinity
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
2a7459306ef18a84caf37b17b4f18c15e3843e9a8ff092a9ea6d298e88a25f8c
+ docker exec -it uniffi-generators uniffi-bindgen generate src/libtelio.udl --language python
+ [[ --restart == \-\-\r\e\s\t\a\r\t ]]
+ docker container kill libtelio_linux_builder
Error response from daemon: Cannot kill container: libtelio_linux_builder: No such container: libtelio_linux_builder
+ echo 'Failed to kill container libtelio_linux_builder!'
Failed to kill container libtelio_linux_builder!
+ docker container rm libtelio_linux_builder
Error response from daemon: No such container: libtelio_linux_builder
+ echo 'Failed to remove container libtelio_linux_builder!'
Failed to remove container libtelio_linux_builder!
+ docker run --name=libtelio_linux_builder -d -v /Users/tomaszklak/Development/libtelio-build/ci/..:/libtelio-build -v /Users/tomaszklak/.ssh:/root/.ssh:ro --workdir=/libtelio-build ghcr.io/nordsecurity/build-linux-rust1.72.1:v0.0.1 sleep infinity
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
5434a229d8ddc59c3690bbd7c4ad60f677e6512e9e67c640ee97969075d2fd87
+ docker exec -it libtelio_linux_builder git config --global --add safe.directory '*'
+ '[' -n '' ']'
+ docker exec -it libtelio_linux_builder git config --global url.ssh://git@bucket.digitalarsenal.net.insteadOf https://bucket.digitalarsenal.net
+ docker exec -it libtelio_linux_builder git config --global --add url.ssh://git@bucket.digitalarsenal.net.insteadOf git://bucket.digitalarsenal.net
+ docker exec -it --env ARCH=arm64 --env CI_JOB_TOKEN= --env LIBTELIO_ENV_SEC_GITLAB_REPOSITORY=bucket.digitalarsenal.net --env 'GIT_SSH_COMMAND=ssh -o StrictHostKeyChecking=no' libtelio_linux_builder libtelio/ci/build_libtelio.py build linux arm64 uniffi-test-bindings
usage: build_libtelio.py [-h] {build,lipo,xcframework,aar,build-ios-simulator-stubs,build-tvos-simulator-stubs} ...
build_libtelio.py: error: unrecognized arguments: uniffi-test-bindings
Traceback (most recent call last):
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 201, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 76, in main
    run_build_command("linux", args)
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 113, in run_build_command
    run_command(command)
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 22, in run_command
    subprocess.check_call(command, env=env)
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['../../ci/build.sh', '--default', 'linux', '--uniffi-test-bindings']' returned non-zero exit status 2.
```

Reply:
The problem is still there for me:
```
################################################################################
Project root (/Users/tomaszklak/Development/libtelio-build) and 'triggered-ref' in ../.github/workflows/gitlab.yml differ
        'cabf069d021d0183e95069ea0617a9ce0ecc72ec' vs 'v1.0.3'
!!! Nat-lab might not behave correctly !!!
################################################################################
|EXECUTE| ../../ci/build.sh --default linux --uniffi-test-bindings

...

usage: build_libtelio.py [-h] {build,lipo,xcframework,aar,build-ios-simulator-stubs,build-tvos-simulator-stubs} ...
build_libtelio.py: error: unrecognized arguments: uniffi-test-bindings
Traceback (most recent call last):
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 201, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 76, in main
    run_build_command("linux", args)
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 113, in run_build_command
    run_command(command)
  File "/Users/tomaszklak/Development/libtelio-build/libtelio/nat-lab/./run_local.py", line 22, in run_command
    subprocess.check_call(command, env=env)
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['../../ci/build.sh', '--default', 'linux', '--uniffi-test-bindings']' returned non-zero exit status 2.
```

Reply:
Yes, it's no longer happening for me.